### PR TITLE
[ISSUE #137]add transaction producer

### PR DESCRIPTION
**Pull request by xujianhai666**
## What is the purpose of the change

#137 add transaction producer

### CODE REVIEW

**Comment by ShannonDing** [comment]
@xujianhai666 Could you please fix the conflicts? it is caused by the CI properties.

**Comment by xujianhai666** [comment]
> @xujianhai666 Could you please fix the conflicts? it is caused by the CI properties.

@ShannonDing done

File: internal/model.go [hunk 1]
```diff
@@ -64,10 +72,79 @@ type consumerData struct {
...
 
+func (c consumerData) UniqueID() string {
+	return c.GroupName
+}
+
+type UniqueItem interface {
+	UniqueID() string
+}
+
+type Set struct {
```
Comment (wenfengwang) [inline internal/model.go]: could remove the data-struct to utils ?

Comment (xujianhai666) [reply internal/model.go]: @wenfengwang  done

**Comment by wenfengwang** [comment]
LGTM

**Pull request merged**